### PR TITLE
Rui/hadamard

### PR DESCRIPTION
**Pull request by bbaayezi**
### Purpose
This PR adds hadamard gate implementation

### Changes
New files: `hadamard-gate/hadamard-gate.html` and `hadamard-gate/hadamard-gate.js`

### CODE REVIEW

**Review by KhesimReid — CHANGES_REQUESTED** [review]
Update hadamard-gate.js to use the Python Shell and the snippets.js file instead of the global script variable, please.

File: quantum/nodes/classical-register/classical-register.js [hunk 1]
```diff
@@ -52,13 +52,15 @@ module.exports = function(RED) {
       };
       flowContext.set('quantumCircuit', structure);
 
-      // Counting the number of registers that were set in the 'structure' array
-      let count = 0;
-      structure.map((x) => {
-        if (typeof(x) !== 'undefined') {
-          count += 1;
-        }
-      });
```
Comment (louislefevre) [inline quantum/nodes/classical-register/classical-register.js:61]: Why is this removed?

Comment (louislefevre) [reply quantum/nodes/classical-register/classical-register.js:61]: Same for quantum-register.js

File: quantum/nodes/hadamard-gate/hadamard-gate.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<script type="text/javascript">
+    RED.nodes.registerType("hadamard-gate", {
+      category: "quantum",
+      color: "#FA4D56",
+      defaults: {
+        name: { value: "H" },
```
Comment (louislefevre) [inline quantum/nodes/hadamard-gate/hadamard-gate.html:6]:
```suggestion
        name: { value: "" },
```
Ideally name default should be left blank.

File: quantum/nodes/hadamard-gate/hadamard-gate.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      icon: "font-awesome/fa-header",
+      align: 'right',
+      paletteLabel: "H",
+      label: function () {
+        return this.name || "H";
+      },
+      inputLabels: function(index) {
+          return "qubit " + index;
+      },
+      outputLabels: "H",
```
Comment (louislefevre) [inline quantum/nodes/hadamard-gate/hadamard-gate.html:19]:
```suggestion
      paletteLabel: "hadamard",
      label: function () {
        return this.name || "hadamard";
      },
      inputLabels: function(index) {
          return "qubit " + index;
      },
      outputLabels: "hadamard",
```
We should have the full name of the gate here, not just a "H".

**Review by KhesimReid — APPROVED** [review]

File: quantum/nodes/hadamard-gate/hadamard-gate.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const util = require('util');
+const snippets = require('../../snippets');
+const shell = require('../../python').PythonShell;
+
+module.exports = function(RED) {
+  function HadamardGate(config) {
```
Comment (louislefevre) [inline quantum/nodes/hadamard-gate/hadamard-gate.js:8]:
```suggestion
  function HadamardGateNode(config) {
```

File: quantum/nodes/hadamard-gate/hadamard-gate.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        `${qrConfig.qubit}`);
+      await shell.execute(shellScript, (err) => {
+        if (err) node.error(err);
+      });
+      // pass the quantum register config to the output
+      send(msg);
+    })
+  }
+
+  RED.nodes.registerType('hadamard-gate', HadamardGate);
```
Comment (louislefevre) [inline quantum/nodes/hadamard-gate/hadamard-gate.js:36]:
```suggestion
  RED.nodes.registerType('hadamard-gate', HadamardGateNode);
```

**Review by louislefevre — CHANGES_REQUESTED** [review]
Small thing but the node functions should end with "Node". Other than that, looks good.

**Review by louislefevre — APPROVED** [review]
